core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java

```diff
@@ -506,8 +508,13 @@ public void write(final String name, final byte[] data) {
                 // Build the config
 
                 final ResultHandle config = carc.checkCast(carc.invokeVirtualMethod(SRCB_BUILD, builder), SmallRyeConfig.class);
-                final ResultHandle providerResolver = carc.newInstance(SCPR_CONSTRUCT, config);
-                carc.invokeStaticMethod(CPR_SET_INSTANCE, providerResolver);
+
+                // ConfigProviderResolver.setInstance(new SimpleConfigurationProviderResolver())
+                carc.invokeStaticMethod(CPR_SET_INSTANCE,
+                        carc.newInstance(MethodDescriptor.ofConstructor(SimpleConfigurationProviderResolver.class)));
```

Comment:
The method descriptor should be a constant like the others.

Reply:
Descriptors constants  are IMO useless when only used once. But I can change this..